### ISSUE TITLE
VQEClient options backend name problem

### ISSUE THREAD

**Issue by jvscursulim** [Qiskit/qiskit-nature#1100]
### Environment

- **Qiskit Nature version**: 0.5.2
- **Python version**: 3.9.12
- **Operating system**:  Windows 11

### What is happening?

When I tried to run a job using VQEClient I received the error message below

Traceback (most recent call last):
  Cell In[10], line 1
    vqe.compute_minimum_eigenvalue(hamiltonian)
  File /opt/conda/lib/python3.10/site-packages/qiskit_nature/runtime/vqe_client.py:292 in compute_minimum_eigenvalue
    options = {"backend_name": self.backend.name()}
TypeError: 'str' object is not callable

Use %tb to get the full traceback.

### How can we reproduce the issue?

```python
import numpy as np

from qiskit.circuit.library import TwoLocal
from qiskit.opflow import Z
from qiskit.algorithms.optimizers import SPSA
from qiskit_ibm_provider import IBMProvider
from qiskit_nature.runtime import VQEClient

provider = IBMProvider(instance="ibm-q/open/main")
backends = provider.backends()

hamiltonian = Z ^ Z
ansatz = TwoLocal(num_qubits=2, rotation_blocks=["ry"], entanglement="linear")
optimizer = SPSA(maxiter=100)
initial_point = np.random.random(ansatz.num_parameters)

vqe = VQEClient(
                ansatz=ansatz,
                optimizer=optimizer,
                provider=provider,
                backend=backends[0],
                initial_point=None,
                shots=1024,
                measurement_error_mitigation=True,
                store_intermediate=True,
                )

vqe.compute_minimum_eigenvalue(hamiltonian)
```

### What should happen?

It is expected that a job will be submitted to the runtime and after queue and running process we should receive the minimum eigenvalue.

### Any suggestions?

The line 292 of vqe_client.py (`options = {"backend_name": self.backend.name()}`) could be replaced by `options = {"backend_name": self.backend.name}`, once that name now is an attribute (there were some changes in qiskit-ibm-provider).

**Comment by Cryoris** [comment]
Yeah, we need to add a case distinction for `BackendV1` (e.g. from the `IBMQ` provider) and `BackendV2` (used by `IBMProvider`), the latter has name only as attribute where the first has it as method 🙂
